tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java

```diff
@@ -144,159 +127,126 @@ public void testInvalidTomlFile() throws BallerinaTestException {
-        errorLeecher.waitForText(5000);
+                new LogLeecher[]{errorLog}, projectPath.toString());
+        errorLog.waitForText(5000);
     }
 
-    @Test
-    public void testRequiredFieldNotFound() throws BallerinaTestException {
-        Path projectPath = Paths.get(negativeTestFileLocation, "missingRequiredField").toAbsolutePath();
-        LogLeecher errorLeecher =
-                new LogLeecher("Value not provided for non-defaultable required field 'username' of " +
-                                       "record 'main:AuthInfo' in configurable variable 'testUser'", ERROR);
-        bMainInstance.runMain("run", new String[]{"main"}, null, new String[]{},
-                              new LogLeecher[]{errorLeecher}, projectPath.toString());
-        errorLeecher.waitForText(5000);
+    @Test(dataProvider = "negative-tests")
+    public void testNegativeCases(String tomlFileName, String errorMsg) throws BallerinaTestException {
+        Path projectPath = Paths.get(negativeTestFileLocation, "configProject").toAbsolutePath();
+        Path tomlPath = Paths.get(negativeTestFileLocation, "config_files", tomlFileName  + ".toml").toAbsolutePath();
+        LogLeecher errorLog = new LogLeecher(errorMsg, ERROR);
+        bMainInstance.runMain("run", new String[]{"main"}, addEnvVariables(tomlPath.toString()), new String[]{},
+                new LogLeecher[]{errorLog}, projectPath.toString());
+        errorLog.waitForText(5000);
     }
 
-    @Test
-    public void testTableKeyNotFound() throws BallerinaTestException {
-        Path projectPath = Paths.get(negativeTestFileLocation, "missingTableKey").toAbsolutePath();
-        LogLeecher errorLeecher =
-                new LogLeecher("Value required for key 'username' of type " +
-                                       "'table<(main:AuthInfo & readonly)> key(username) & readonly'" +
-                                       " in configurable variable 'users'", ERROR);
-        bMainInstance.runMain("run", new String[]{"main"}, null, new String[]{},
-                              new LogLeecher[]{errorLeecher}, projectPath.toString());
-        errorLeecher.waitForText(5000);
-    }
-
-    @Test
-    public void testUnsupportedMap() throws BallerinaTestException {
-        Path projectPath = Paths.get(negativeTestFileLocation, "invalidMapType").toAbsolutePath();
-        LogLeecher errorLeecher =
-                new LogLeecher("Configurable feature is yet to be supported for type 'map<int> & readonly'", ERROR);
-        bMainInstance.runMain("run", new String[]{"main"}, null, new String[]{},
-                              new LogLeecher[]{errorLeecher}, projectPath.toString());
-        errorLeecher.waitForText(5000);
+    @DataProvider(name = "negative-tests")
+    public Object[][] getNegativeTests() {
+        return new Object[][]{
+                {"no_module_config", "Value not provided for required configurable variable 'main:stringVar'"},
```

Comment:
We need to validate the unnecessary valid TOML content inside the config file in future.

Reply:
Created issue #28732